cores/arduino/Uart.cpp

```diff
@@ -91,8 +91,11 @@ void Uart::flush()
 void Uart::IrqHandler()
 {
   if (sercom->availableDataUART()) {
-    rxBuffer.store_char(sercom->readDataUART());
-
+    sercom->isFrameErrorUART();
+    uint8_t c=sercom->readDataUART();
+    if( !sercom->isUARTError()){
```

Comment:
I think this should also check the error type here, from the data sheet not all errors will have data associated with them.